src/pydargs/__init__.py

```diff
@@ -73,10 +92,17 @@ def _create_parser(tp: Type[Dataclass], **kwargs: Any) -> ArgumentParser:
 
 
 def _add_arguments(parser: ArgumentParser, tp: Type[Dataclass], prefix: str = "") -> ArgumentParser:
-    parser_or_group = parser.add_argument_group(prefix.strip("_")) if prefix else parser
+    parser_or_group = (  # Only add a group if there is a prefix that isn't "" or "+".
+        parser.add_argument_group(prefix.strip("_")) if prefix.replace("+", "") else parser
+    )
+    has_subparser = False
     for field in fields(tp):
         if field.metadata.get("ignore_arg", False):
             continue
+        if _is_command(field):
```

Comment:
Stuff like this makes me wonder if we should take a more object oriented approach and start off by interpreting the fields and casting them to our own internal Field types? 

This one add_arguments function is become very complex with all the if-elses, prefixes and string replacements etc. 

Out of scope for this PR, but do you agree?

Reply:
I completely agree that the `add_arguments` function could use some clean-up. Especially all the separate calls to `parser.add_argument` should be refactored to a single one. But inherently we are doing something that is a little complicated...

I'm not sure that defining our own types would help. Perhaps it would help to define more of these helper functions that increase readability?

Anyway, for another PR!

Reply:
I was thinking something along the lines of:

Create a function that essentially does the type checking and creates the correct updates. 

e.g. 
```if origin is Sequence or origin is list:
     return ListField(field)
```

and then we extract all this logic that modifies the argument_kwargs to these classes.

Then we could loop over the fields first and get a list of fields that know about their own state, like whether it has a prefix and metadata etc. 
Then in the parser we just call something like custom_field.get_argument() and this contains whatever we now put in add_argument(...)

but I didn't work this thought out enough to see limitations or whether or not it actually makes it easier 😜

Reply:
I'm not sure that would make things easier. Anyway - let's move this discussion elsewhere as it is not directly related to this PR